src/main/java/seedu/duke/Storage.java

```diff
@@ -0,0 +1,122 @@
+            try {
+                temp.createNewFile();
+                storageFile = temp;
+            } catch (Exception e1) {
+                e1.getStackTrace();
+            }
+        } else {
+            storageFile = temp;
+        }
+    }
+
+    /**
+     * Loads data from data file.
+     * Parses data to list of tasks.
+     * If no data in data file, gives an empty list.
+     *
+     * @return List of stored tasks.
+     */
+    public List<Task> load() {
+        List<Task> result = new ArrayList<>();
+        Scanner storage;
+        try {
+            storage = new Scanner(storageFile);
+
+            while (storage.hasNext()) {
+                String type = storage.nextLine();
+                boolean status = storage.nextLine().equals("1");
+                String content = storage.nextLine();
+                if (type.equals("T")) {
+                    result.add(new Todo(status, content));
+                } else if (type.equals("E")) {
+                    String time = storage.nextLine().replace(" ", "");
+                    result.add(new Event(status, content, time));
+                } else if (type.equals("D")) {
+                    String time = storage.nextLine().replace(" ", "");
+                    result.add(new Deadline(status, content, time));
+                }
+            }
+            storage.close();
+        } catch (FileNotFoundException e) {
+
+        }
+        return result;
+    }
+
+    /**
+     * Saves list of tasks into data file.
+     *
+     * @param taskList List of tasks.
+     */
+    public void save(List<Task> taskList) {
+        try {
+            FileWriter fileWriter = new FileWriter(storageFile);
+
+            String data = "";
+            for (int i = 0; i < taskList.size(); i++) {
+                String temp = "";
+                if (taskList.get(i) instanceof Todo) {
+                    Todo holding = (Todo) taskList.get(i);
+                    temp += "T\n";
+                    temp += (holding.isDone() ? "1" : "0") + "\n";
+                    temp += holding.getContent() + "\n";
+                    data += temp;
+                    temp = "";
+                } else if (taskList.get(i) instanceof Event) {
+                    Event holding = (Event) taskList.get(i);
+                    temp += "E\n";
+                    temp += (holding.isDone() ? "1" : "0") + "\n";
+                    temp += holding.getContent() + "\n";
+                    temp += holding.getTime() + "\n";
+                    data += temp;
+                    temp = "";
+                } else if (taskList.get(i) instanceof Deadline) {
+                    Deadline holding = (Deadline) taskList.get(i);
+                    temp += "D\n";
+                    temp += (holding.isDone() ? "1" : "0") + "\n";
+                    temp += holding.getContent() + "\n";
+                    temp += holding.getTime() + "\n";
+                    data += temp;
+                    temp = "";
```

Comment:
Perhaps it would be possible to concatenate the variable temp altogether?

Reply:
Yeah, I have tried to simplify my codes